src/shared/store/ApiStore/ApiStore.ts

```diff
@@ -0,0 +1,42 @@
+import { ApiResponse, IApiStore, RequestParams } from "./types";
+import qs from "qs";
+
+export default class ApiStore implements IApiStore {
+  readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    // TODO: Примите из параметров конструктора baseUrl
+    // и присвойте его в this.baseUrl
+    this.baseUrl = baseUrl;
+  }
+
+  request<SuccessT, ErrorT = any, ReqT = {}>(
+    params: RequestParams<ReqT>
+  ): Promise<ApiResponse<SuccessT, ErrorT>> {
+    // TODO: Напишите здесь код, который с помощью fetch будет делать запрос
+    let url =
```

Comment:
лучше воспользоваться шаблонными строками
```suggestion
    let url = `${this.baseUrl}${params.endpoint}${params.method === HTTPMethod.GET ? `?${qs.stringify(params.data)}` :  ''}`
```